app/serializers/homepage/article_serializer.rb

```diff
@@ -0,0 +1,41 @@
+module Homepage
+  class ArticleSerializer < ApplicationSerializer
+    attributes(
+      :class_name,
+      :cloudinary_video_url,
+      :comments_count,
+      :id,
+      :path,
+      :public_reactions_count,
+      :readable_publish_date,
+      :reading_time,
+      :title,
+      :user_id,
+    )
+
+    attribute :video_duration_string, &:video_duration_in_minutes
+    attribute :published_at_int, ->(article) { article.published_at.to_i }
+    attribute :tag_list, ->(article) { article.cached_tag_list.to_s.split(", ") }
+    attribute :tag_flare, ->(article, params) { params[:tag_flares][article.id] }
+
+    attribute :user do |article|
+      user = article.user
+
+      {
+        name: user.name,
+        profile_image_90: user.profile_image_90,
+        username: user.username
+      }
+    end
+
+    attribute :organization, if: proc { |a| a.organization.present? } do |article|
+      organization = article.organization
+
+      {
+        name: organization.name,
+        profile_image_90: article.organization.profile_image_90,
+        slug: organization.slug
+      }
+    end
```

Comment:
We're using JSON:API serializers, right? Asking because this isn't how JSON:API serializes associated entities.

Reply:
Yes and no. We use the JSON API serializers because the gem is fast, not because it serializes in JSON API format. That's why the formerly known `fast_jsonapi` library was added back then when the Elasticsearch project started. Additionally, for this project we can't change the serialization format, as we're constrained by how the frontend expects data to be. 

Once Elasticsearch is gone and feature flags are removed, we will be able to refactor both backend and frontend as we please